core/src/callstate.js

```diff
@@ -1,37 +1,39 @@
-import daily from "./core.js";
-
 let state = {};
 let updateHandlers = [];
-let call = {};
 let waitingForSync = false;
+let call = null;
 const SYNC_MAX_WAITING_TIME = 7000;
 const UPDATE_MSG = "update-call-state";
 const REQUEST_MSG = "request-call-state";
 
 /* Public interface */
 export default {
-  onCallStateUpdate: function (key, callback) {
-    if (!updateHandlers[key]) {
-      updateHandlers[key] = [];
-    }
-    updateHandlers[key].push(callback);
-  },
+  afterCreateFrame,
+  instanceMethods: {
+    onCallStateUpdate: function (key, callback) {
+      if (!updateHandlers[key]) {
+        updateHandlers[key] = [];
+      }
+      updateHandlers[key].push(callback);
+    },
 
-  updateCallState: function (key, newState, broadcast = true) {
-    // used for an extension to request a state update,
-    // which will also get broadcast to everybody else on the call.
-    applyStateUpdate(key, newState);
-    if (broadcast) {
-      broadcastStateUpdate(key, newState);
-    }
-  },
+    updateCallState: function (key, newState, broadcast = true) {
+      // used for an extension to request a state update,
+      // which will also get broadcast to everybody else on the call.
+      console.log("Updating call state:", newState);
+      applyStateUpdate(key, newState);
+      if (broadcast) {
+        broadcastStateUpdate(key, newState);
+      }
+    },
 
-  state,
+    callState: state,
```

Comment:
Looking through the code, this seems to be only referenced in the iFrame extension. And in that extension, it's only being used to populate the iFrame state. Do we need a callstate at all? Or should this all just be in the iFrame extension? In what other extension situations would we need callstate that aren't using an iFrame?

Just trying to figure out a way to keep the API surface level as small as possible 🤔